rootfs-builder/rootfs.sh

```diff
@@ -381,6 +381,13 @@ echo "refclock PHC /dev/ptp0 poll 3 dpoll -2 offset 0" >> ${chrony_conf_file}
 # Reference:  https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html
 sed -i 's/^\(server \|pool \|peer \)/# &/g'  ${chrony_conf_file}
 
+chrony_systemd_service="${ROOTFS_DIR}/usr/lib/systemd/system/chronyd.service"
+if [ ${distro} == ubuntu ] || [ ${distro} == debian ] ; then
+	chrony_systemd_service="${ROOTFS_DIR}/lib/systemd/system/chrony.service"
+fi 
+
+sed -i '/^\[Unit\]/a ConditionPathExists=\/dev\/ptp0' ${chrony_systemd_service}
```

Comment:
Responding to @jcvenegas's thought about creating our own unit, I think the way systemd expects use to handle this is by using a snippet. Something like:

```sh
$ sudo mkdir -p /etc/systemd/system/chronyd.service.d/
$ cat <<EOT | sudo tee /etc/systemd/system/chronyd.service.d/condition-path-exists.conf
[Unit]
ConditionPathExists=/dev/ptp0
EOT
```

Reply:
Nice alternative, just wonder if this could be distro agnostic better! :)